File: b2/2798/2798.cpp
#include<iostream>
using namespace std;

int main()
{
	int n, m;
	cin >> n >> m;
	int *cards = new int[n];
	for(int i=0; i<n; i++)
		cin >> cards[i];
	int result = 0;
	for(int i=0; i<n-2; i++)
		for(int j=i+1; j<n-1; j++)
			for(int k=j+1; k<n; k++)
			{
				int temp = cards[i]+cards[j]+cards[k];
				if(temp>result && temp <= m)
					result = temp;
			}
	cout << result << "\n";

	delete[] cards;
	return 0;
}

File: b2/2798/2798.py
n, m = map(int, input().split())
cards = sorted(list(map(int, input().split())))
result = 0
for i in range(n-2):
    for j in range(i+1, n-1):
        for k in range(j+1, n):
            temp = cards[i] + cards[j] + cards[k]
            if temp > m:
                break
            elif temp > result:
                result = temp
print(result)
